### PR TITLE
BE-4-add-user-model

### PR DESCRIPTION
**Pull request by Randayy**
added models.py with User Model.
added schemes.py with needed schemes for User.
made migration 'initial_migration' which created table User.
table is correctly created , I attached screenshot above.
telegram-cloud-photo-size-2-5249191729775637543-x

added simple logging , I attached screenshot above.
telegram-cloud-photo-size-2-5249191729775637542-y

### CODE REVIEW

File: app/db/connect_postgresql.py
Comment (Randayy) [inline app/db/connect_postgresql.py:1]: чомусь не працювало на env host

Comment (Artemooon) [reply app/db/connect_postgresql.py:1]: если ты через докер компоуз стартуешь, то нужно указывать имя контейнера с постгрес как хост (типо postgres)

File: app/db/connect_postgresql.py [hunk 1]
```diff
@@ -6,9 +6,10 @@
 from app.core.config import Settings
 from sqlalchemy import text
 
+
 settings = Settings()
 
-DATABASE_URL = f"postgresql+asyncpg://{settings.db_user}:{settings.db_password}@{settings.db_host}:{settings.db_port}/{settings.db_name}"
+DATABASE_URL = f"postgresql+asyncpg://{settings.db_user}:{settings.db_password}@localhost:{settings.db_port}/{settings.db_name}"
```
Comment (Artemooon) [inline app/db/connect_postgresql.py]: fix this pls

File: app/db/models.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from sqlalchemy import Column, Integer, String
+from sqlalchemy.ext.declarative import declarative_base
+
+Base = declarative_base()
+
+
+class User(Base):
```
Comment (Artemooon) [inline app/db/models.py]: move id field to the Base model and then you will not need to specify id for each model

File: app/db/connect_postgresql.py
Comment (Randayy) [reply app/db/connect_postgresql.py:1]: Хорошо , працює)

File: app/db/connect_postgresql.py (hunk 1 above)
Comment (Randayy) [reply app/db/connect_postgresql.py]: fixed

File: app/db/models.py (hunk 2 above)
Comment (Randayy) [reply app/db/models.py]: fixed 
telegram-cloud-photo-size-2-5253781778504736290-y

File: alembic/versions/281c0324d7c4_adding_basemodel.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    pass
+    # ### end Alembic commands ###
+
+
+def downgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    pass
+    # ### end Alembic commands ###
```
Comment (Artemooon) [inline alembic/versions/281c0324d7c4_adding_basemodel.py]: why do we need an empty migration?

File: app/db/models.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from sqlalchemy import Column, Integer, String
+from sqlalchemy.ext.declarative import declarative_base
+
+Base = declarative_base()
+
+class BaseTable(Base):
+    __abstract__ = True
+
+    id = Column(Integer, primary_key=True, index=True)
```
Comment (Artemooon) [inline app/db/models.py]: create a base_models.py file and move these models there

File: app/db/models.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+class User(BaseTable):
+    __tablename__ = 'users'
+
+    username = Column(String, unique=True, index=True)
+    password = Column(String)
+    email = Column(String, unique=True, index=True)
+    first_name = Column(String, nullable=True)
+    last_name = Column(String, nullable=True)
+    full_name = Column(String, nullable=True)
```
Comment (Artemooon) [inline app/db/models.py]: no need to keep the extra column which store full_name, this is the waste of the disk space, you can just create a property which will concat first_name and last_name in the model

File: app/schemas/schemas.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class User(UserBase):
+    id: int
+
+    class Config:
+        orm_mode = True
+
+
+class SignInRequestModel(BaseModel):
```
Comment (Artemooon) [inline app/schemas/schemas.py]: why do you use Model ending here but not in the models above?
use the consistent naming please 
UserBaseSchema, UserCreateSchema, UserSchema etc...
or
UserUpdateRequest, UserUpdateResponse, UserCreateRequest etc...

File: app/schemas/schemas.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    password: str
+
+
+class SignUpRequestModel(UserCreate):
+    confirm_password: str
+
+
+class UserUpdateRequestModel(UserUpdate):
+    current_password: str
+    new_password: str
```
Comment (Artemooon) [inline app/schemas/schemas.py:35]: why do you need the separate schema here?

Comment (Randayy) [reply app/schemas/schemas.py:35]: deleted this

File: app/schemas/schemas.py (hunk 6 above)
Comment (Randayy) [reply app/schemas/schemas.py]: fixed

File: app/db/models.py (hunk 4 above)
Comment (Randayy) [reply app/db/models.py]: moved

File: alembic/versions/281c0324d7c4_adding_basemodel.py (hunk 3 above)
Comment (Randayy) [reply alembic/versions/281c0324d7c4_adding_basemodel.py]: deleted and created normal migration

File: app/db/models.py
Comment (Artemooon) [inline app/db/models.py:1]: change file name to the user_models.py

File: app/schemas/schemas.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
+from pydantic import BaseModel, EmailStr
+from typing import List, Optional
+
+from pydantic import BaseModel, EmailStr, Field
+
+
+class UserBase(BaseModel):
```
Comment (Artemooon) [inline app/schemas/schemas.py]: use Schema ending, like UserBaseSchema, UserCreateSchema, UserSchema, for all the models

File: alembic/versions/ab44aecbe779_initial.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    sa.Column('email', sa.String(), nullable=True),
+    sa.Column('first_name', sa.String(), nullable=True),
+    sa.Column('last_name', sa.String(), nullable=True),
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.create_index(op.f('ix_users_email'), 'users', ['email'], unique=True)
+    op.create_index(op.f('ix_users_id'), 'users', ['id'], unique=False)
+    op.create_index(op.f('ix_users_username'), 'users', ['username'], unique=True)
+    # ### end Alembic commands ###
```
Comment (Artemooon) [inline alembic/versions/ab44aecbe779_initial.py]: remove comments "please adjust" and also leave only single migration, you have 3 migrations now that all together apply changes to the User model. Just leave single initial migrations that create users table

File: app/schemas/schemas.py (hunk 8 above)
Comment (Randayy) [reply app/schemas/schemas.py]: fixed

File: alembic/versions/ab44aecbe779_initial.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+down_revision: Union[str, None] = '978c76696bff'
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('users',
+    sa.Column('username', sa.String(), nullable=True),
+    sa.Column('password', sa.String(), nullable=True),
```
Comment (Artemooon) [inline alembic/versions/ab44aecbe779_initial.py]: why nullable=True for the password? it should be required , username also should be required, change it in the models as well

File: alembic/versions/ab44aecbe779_initial.py (hunk 9 above)
Comment (Randayy) [reply alembic/versions/ab44aecbe779_initial.py]: removed and there is now only 1 migration

File: alembic/versions/ab44aecbe779_initial.py (hunk 10 above)
Comment (Randayy) [reply alembic/versions/ab44aecbe779_initial.py]: changed , i thought i will change it when i will add data and authorization

File: alembic/versions/f6b91c1bac43_initial.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+down_revision: Union[str, None] = None
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.create_table('users',
+    sa.Column('username', sa.String(), nullable=False),
+    sa.Column('password', sa.String(), nullable=False),
+    sa.Column('email', sa.String(), nullable=True),
```
Comment (Artemooon) [inline alembic/versions/f6b91c1bac43_initial.py]: nullable=False

Comment (Randayy) [reply alembic/versions/f6b91c1bac43_initial.py]: в email ?

**Review by Artemooon — APPROVED** [review]

**Pull request merged**